Bottom: a77d213374ac50c2f72dbf7097437d4fa9c164ce
Top:    bed49600b089c286fcacfc8a1fc699be67e6deae
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-03 23:35:55 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index af92d10..13d0dc3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -230,6 +230,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
